testtools/iothub_test/src/iothub_account.c

```diff
@@ -976,6 +976,19 @@ IOTHUB_PROVISIONED_DEVICE* IoTHubAccount_GetX509Device(IOTHUB_ACCOUNT_INFO_HANDL
     }
 }
 
+IOTHUB_PROVISIONED_DEVICE* IoTHubAccount_GetDevice(IOTHUB_ACCOUNT_INFO_HANDLE acctHandle, IOTHUB_ACCOUNT_AUTH_METHOD accountAuthMethod)
+{
+    if (accountAuthMethod == IOTHUB_ACCOUNT_AUTH_X509)
+    {
+        return IoTHubAccount_GetX509Device(acctHandle);
+    }
+    else
+    {
+        return IoTHubAccount_GetSASDevice(acctHandle);
+    }
+}
```

Comment:
return NULL;

Reply:
please note that IoTHubAccount_GetX509Device() and IoTHubAccount_GetSASDevice() can return NULL....but it looks like our code does not handle this.

Reply:
Making change as suggested along with the change above with explict test for ==.  

The test basically use an "exception" based approach where if anything goes wrong, we kick an `ASSERT` right away.  We don't check against NULL because the test wouldn't have got to that state anyway.  The test init in this instance does a `dt_e2e_init` and if this doesn't setup devices we won't execute any further tests.